### PR TITLE
ENT-10897: master deps updates for 3.23.0 release

### PR DESCRIPTION
**Pull request by craigcomstock**
- Updated lmdb from 0.9.30 to 0.9.31
- Updated zlib from 1.2.13 to 1.3
- Updated openssl from 3.1.1 to 3.1.4
- Updated libxml2 from 2.11.4 to 2.11.5
- Updated openldap from 2.6.4 to 2.6.6
- Updated libcurl-hub from 8.1.2 to 8.4.0
- Updated apache from 2.4.57 to 2.4.58
- Updated git from 2.41.0 to 2.42.1
- Updated postgresql from 15.3 to 15.5
- Updated php from 8.2.7 to 8.2.12
- Updated libcurl from 8.1.2 to 8.4.0

### CODE REVIEW

**Comment by craigcomstock** [comment]
@cf-bottom jenkins with exotics, should all work now.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10060/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10060/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10060/

File: deps-packaging/libcurl/hpux/build [hunk 1]
```diff
@@ -28,7 +31,7 @@ TTD=${BUILD_ROOT}/cfbuild-libcurl-devel${PREFIX}
     --without-polarssl \
     --without-winidn \
     --without-winssl \
-    CPPFLAGS="-DAF_LOCAL=AF_UNIX"
+    CPPFLAGS="-DAF_LOCAL=AF_UNIX -D_LARGEFILE_SOURCE"
```
Comment (craigcomstock) [inline deps-packaging/libcurl/hpux/build:34]: @larsewi @vpodzime I tried many variations of `-D_GNU_SOURCE=1` (which fails to set `_LARGEFILE_SOURCE` as described [here](https://www.man7.org/linux/man-pages/man7/feature_test_macros.7.html) probably due to the age of gcc on this hpux host (4.2.3)) so ended up with this option.

I ran a portion of libcurl's 1500 some tests (about half) and they seemed to work alright as a way to check that things worked and didn't just compile.

**Review by larsewi — APPROVED** [review]

**Comment by craigcomstock** [comment]
mingw failure was a flake, retry OK: [Build Status](https://ci.cfengine.com/job/pr-pipeline/10062/)

**Pull request merged**